### PR TITLE
Accelsim/trace feed

### PR DESCRIPTION
**Pull request by gongchen618**
fix some structural issues

### CODE REVIEW

File: accelsim_tracing/driver/driver.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+)
+
+type Driver struct {
+	gpu *gpu.GPU
+}
+
+func (d *Driver) Exec(bm *Benchmark) error {
+	if bm == nil {
+		return errors.New("no trace parser specified")
+	} else if d.gpu == nil {
```
Comment (syifan) [inline accelsim_tracing/driver/driver.go:17]: If return in the previous case, no need to use `else`.

File: accelsim_tracing/driver/driver.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+			tgReader.ParseThreadBlocks()
+			for it := tgReader.ThreadBlockQueue.Front(); it != nil; it = it.Next() {
+				d.gpu.RunThreadBlock(it.Value.(*trace.ThreadBlock).GenerateNVThreadBlock())
+			}
+
+			tgReader.File.Close()
+		}
+	}
+	return nil
```
Comment (syifan) [inline accelsim_tracing/driver/driver.go:36]: Empty line before return and after for loop.

File: accelsim_tracing/driver/benchmark_builder.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	b.traceDirPath = path
+	b.fromTrace = true
+	return b
+}
+
+func (b *BenchmarkBuilder) Build() (*Benchmark, error) {
+	if !b.fromTrace {
+		return nil, errors.New("no trace dir path specified")
+	}
+	bm := &Benchmark{
```
Comment (syifan) [inline accelsim_tracing/driver/benchmark_builder.go:31]: Add an empty line after the closing of every bracket.

File: accelsim_tracing/smunit/smunit.go [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+	registerFile *RegisterFile
+	aluInt32     []alu.ALU
+}
+
+func (s *SMUnit) Execute(warp *nvidia.Warp) {
+
+}
+
+func (s *SMUnit) IsFree() bool {
+	return true
```
Comment (shawlleyw) [inline accelsim_tracing/smunit/smunit.go:18]: Limit max number of warps

File: accelsim_tracing/driver/driver_builder.go [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+type DriverBuilder struct {
+	gpu *gpu.GPU
+}
+
+func NewDriverBuilder() *DriverBuilder {
+	return &DriverBuilder{
+		gpu: nil,
+	}
+}
+func (d *DriverBuilder) WithGPU(g *gpu.GPU) *DriverBuilder {
```
Comment (syifan) [inline accelsim_tracing/driver/driver_builder.go:14]: Empty line before function.

**Review by syifan — APPROVED** [review]

File: accelsim_tracing/sm/sm.go [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package sm
+
+import (
+	"github.com/sarchlab/mgpusim/v3/accelsim_tracing/nvidia"
+	"github.com/sarchlab/mgpusim/v3/accelsim_tracing/smunit"
+)
+
+type SM struct {
```
Comment (shawlleyw) [inline accelsim_tracing/sm/sm.go:8]: SM is_free: limit the max number of blocks

**Review by shawlleyw — COMMENTED** [review]
Reminder

**Pull request closed without merging**